From 91cd747a8e5bb8a8a90cef52781c70cdf2d1958f Mon Sep 17 00:00:00 2001
From: Sam Mortimer <sam@mortimer.me.uk>
Date: Thu, 27 Sep 2018 22:17:17 -0700
Subject: [PATCH 6/6] lineage-sdk: Move force show navbar to lineage system
 settings

*) Mild cleanup of DEV_FORCE_SHOW_NAVBAR in the process.

Change-Id: I519245ffb350c172bb1eef2b0d507089fc87907c
---
 .../res/values/defaults.xml                   |  2 +-
 .../LineageSettingsProviderDefaultsTest.java  |  6 ++---
 .../lineageos/providers/LineageSettings.java  | 23 +++++++++++--------
 3 files changed, 17 insertions(+), 14 deletions(-)

diff --git a/packages/LineageSettingsProvider/res/values/defaults.xml b/packages/LineageSettingsProvider/res/values/defaults.xml
index caeff09..3a23f52 100644
--- a/packages/LineageSettingsProvider/res/values/defaults.xml
+++ b/packages/LineageSettingsProvider/res/values/defaults.xml
@@ -62,7 +62,7 @@
          $1=MODEL -->
     <string name="def_device_name">%1$s</string>
 
-    <!-- Defaults for LineageSettings.Global.DEV_FORCE_SHOW_NAVBAR -->
+    <!-- Defaults for LineageSettings.System.FORCE_SHOW_NAVBAR -->
     <integer name="def_force_show_navbar">0</integer>
 
     <!-- Default for LineageSettings.Global.HEADS_UP_NOTIFICATIONS_ENABLED
diff --git a/packages/LineageSettingsProvider/tests/src/org/lineageos/lineagesettings/tests/LineageSettingsProviderDefaultsTest.java b/packages/LineageSettingsProvider/tests/src/org/lineageos/lineagesettings/tests/LineageSettingsProviderDefaultsTest.java
index f5c7ba3..fee6742 100644
--- a/packages/LineageSettingsProvider/tests/src/org/lineageos/lineagesettings/tests/LineageSettingsProviderDefaultsTest.java
+++ b/packages/LineageSettingsProvider/tests/src/org/lineageos/lineagesettings/tests/LineageSettingsProviderDefaultsTest.java
@@ -79,6 +79,9 @@ public class LineageSettingsProviderDefaultsTest extends AndroidTestCase {
         SYSTEM_SETTINGS_DEFAULTS.add(new Setting(
                 LineageSettings.System.STATUS_BAR_BATTERY_STYLE,
                 "R.integer.def_battery_style"));
+        SYSTEM_SETTINGS_DEFAULTS.add(new Setting(
+                LineageSettings.System.FORCE_SHOW_NAVBAR,
+                "R.integer.def_force_show_navbar"));
     }
 
     //SECURE
@@ -114,9 +117,6 @@ public class LineageSettingsProviderDefaultsTest extends AndroidTestCase {
         GLOBAL_SETTINGS_DEFAULTS.add(new Setting(
                 LineageSettings.Global.WEATHER_TEMPERATURE_UNIT,
                 "R.integer.def_temperature_unit"));
-        GLOBAL_SETTINGS_DEFAULTS.add(new Setting(
-                LineageSettings.Global.DEV_FORCE_SHOW_NAVBAR,
-                "R.integer.def_force_show_navbar"));
     }
 
     @Override
diff --git a/sdk/src/java/lineageos/providers/LineageSettings.java b/sdk/src/java/lineageos/providers/LineageSettings.java
index fd88771..17a81c9 100644
--- a/sdk/src/java/lineageos/providers/LineageSettings.java
+++ b/sdk/src/java/lineageos/providers/LineageSettings.java
@@ -463,7 +463,6 @@ public final class LineageSettings {
         protected static final ArraySet<String> MOVED_TO_SECURE;
         static {
             MOVED_TO_SECURE = new ArraySet<>(1);
-            MOVED_TO_SECURE.add(Secure.DEV_FORCE_SHOW_NAVBAR);
         }
 
         // region Methods
@@ -2050,6 +2049,16 @@ public final class LineageSettings {
         public static final Validator LONG_SCREEN_APPS_VALIDATOR =
                 sAlwaysTrueValidator;
 
+        /**
+         * Force show navigation bar setting.
+         * @hide
+         */
+        public static final String FORCE_SHOW_NAVBAR = "force_show_navbar";
+
+        /** @hide */
+        public static final Validator FORCE_SHOW_NAVBAR_VALIDATOR =
+                sBooleanValidator;
+
         /**
          * I can haz more bukkits
          * @hide
@@ -2171,9 +2180,6 @@ public final class LineageSettings {
         public static boolean shouldInterceptSystemProvider(String key) {
             switch (key) {
                 case System.SYSTEM_PROFILES_ENABLED:
-                // some apps still query Settings.System.DEV_FORCE_SHOW_NAVBAR;
-                // we intercept the call, and return LineageSettings.Secure.DEV_FORCE_SHOW_NAVBAR's value
-                case Secure.DEV_FORCE_SHOW_NAVBAR:
                     return true;
                 default:
                     return false;
@@ -2339,6 +2345,8 @@ public final class LineageSettings {
                     ACCELEROMETER_ROTATION_ANGLES_VALIDATOR);
             VALIDATORS.put(LONG_SCREEN_APPS,
                     LONG_SCREEN_APPS_VALIDATOR);
+            VALIDATORS.put(FORCE_SHOW_NAVBAR,
+                    FORCE_SHOW_NAVBAR_VALIDATOR);
             VALIDATORS.put(__MAGICAL_TEST_PASSING_ENABLER,
                     __MAGICAL_TEST_PASSING_ENABLER_VALIDATOR);
         };
@@ -2365,7 +2373,6 @@ public final class LineageSettings {
         protected static final ArraySet<String> MOVED_TO_GLOBAL;
         static {
             MOVED_TO_GLOBAL = new ArraySet<>(1);
-            MOVED_TO_GLOBAL.add(Global.DEV_FORCE_SHOW_NAVBAR);
         }
 
         // region Methods
@@ -3216,11 +3223,6 @@ public final class LineageSettings {
          */
         public static boolean shouldInterceptSystemProvider(String key) {
             switch (key) {
-                // some apps still query Settings.System.DEV_FORCE_SHOW_NAVBAR, and it was moved to
-                // Settings.Secure, then LineageSettings.Secure. Forward queries from Settings.Secure
-                // to LineageSettings.Secure here just in case an app stuck with the Settings.Secure call
-                case DEV_FORCE_SHOW_NAVBAR:
-                    return true;
                 default:
                     return false;
             }
@@ -3655,6 +3657,7 @@ public final class LineageSettings {
 
         /**
          * Developer options - Navigation Bar show switch
+         * @deprecated
          * @hide
          */
         public static final String DEV_FORCE_SHOW_NAVBAR = "dev_force_show_navbar";
-- 
2.17.1

